addons/test_mail/tests/test_mail_thread_internals.py

```diff
@@ -454,6 +454,24 @@ def test_unlink_notification_message(self):
 class TestNoThread(MailCommon, TestRecipients):
     """ Specific tests for cross models thread features """
 
+    @users('employee')
+    def test_mail_sending_on_non_thread_model(self):
+        """ This test simulates scenarios where a required method called `_process_attachments_for_post` is missing,
+        in such case composer should fallback to the method implementation in mail.thread. """
+        template = self.env['mail.template'].create({
+            'name': 'TestTemplate',
+            'model_id': self.env['ir.model']._get_id('mail.test.nothread'),
+        })
+        mail_compose_message = self.env['mail.compose.message'].create({
+            'composition_mode': 'mass_mail',
+            'model': 'mail.test.nothread',
+            'template_id': template.id,
+        })
+        record = self.env['mail.test.nothread'].sudo().create({
+            'name': 'Test Model Missing Method',
+        })
+        mail_compose_message._prepare_mail_values(record.ids)
```

Comment:
Sorry for doing it in several steps, but why not calling the main send method ? That way we don't depend on some implementation details but we test the global flow.